crates/gitbutler-branch-actions/src/remote.rs

```diff
@@ -60,45 +61,51 @@ pub struct RemoteCommit {
     pub conflicted: bool,
 }
 
-/// Return information on all local branches, while skipping gitbutler-specific branches in `refs/heads`.
-///
-/// Note to be confused with `list_branches()`, which is used for the new branch listing.
+/// Finds all branches matching a given name, which can be at most one local branch,
+/// and any number of branches (on different remotes).
 ///
 /// # Previous notes
 /// For legacy purposes, this is still named "remote" branches, but it's actually
 /// a list of all the normal (non-gitbutler) git branches.
-pub fn list_local_branches(ctx: &CommandContext) -> Result<Vec<RemoteBranch>> {
-    let default_target = default_target(&ctx.project().gb_dir())?;
+pub fn find_git_branches(ctx: &CommandContext, branch_name: &str) -> Result<Vec<RemoteBranch>> {
+    let repo = ctx.repository();
+    let remotes_raw = repo.remotes()?;
+    let remotes: Vec<_> = remotes_raw.iter().flatten().collect();
+
+    // We since we are testing for the presence of branches we swallow any errors
+    // from both finding the branch, and looking up the branch commit data. The
+    // latter can fail if/when a ref points to a object that doesn't exist.
+    let mut all_branches: Vec<RemoteBranch> = remotes
+        .iter()
+        .filter_map(|remote_name| {
+            let branch_name = format!("{}/{}", remote_name, branch_name);
+            let blah = repo
+                .find_branch(&branch_name, BranchType::Remote)
+                .ok()
+                .and_then(|branch| branch_to_remote_branch(&branch, &[remote_name]).ok());
+            blah
+        })
+        .collect();
```

Comment:
@Caleb-T-Owens I think it's best if the functions return errors, since one would expect them to work if you are looking up a valid repo and branch. In the case of `get_remote_branch_data` it's natural to fail the front-end request, while here in a "find" function it feels ok to swallow them?

Reply:
Can we talk about this on a call tomorrow?